ERCS/erc-7803.md

```diff
@@ -0,0 +1,157 @@
+
+Requests for typed data signatures via JSON-RPC (`eth_signTypedData`) or client libraries are extended with the following properties:
+
+- `signingDomains`
+- `authMethods`
+
+These new properties are used along the the existing ones, i.e., `types`, `primaryType`, `domain`, and `message`.
+
+The signature returned in response to the request MAY be of any size, and in the absence of `signingDomains` it MUST be treated opaquely as the type of the signature is not known.
+
+### `signingDomains`
+
+This property is an array of the chain of smart contract accounts that the signature is on behalf of, listed from "closest" to "furthest".
+
+Each member of the array is an object with the following keys:
+
+- `types`: An object with the same format as EIP-712 `types` with at least an `EIP712Domain` key.
+- `domain`: An object that matches the type `EIP712Domain` described in `types`.
+
+For example: 
+
+1. A dapp requests an EIP-712 signature to a connected account via JSON-RPC. `signingDomains` is empty or undefined.
+2. The connected account is a multisig, so it requests EIP-712 signatures from its signers, prepending the domain of the multisig to `signingDomains`, which is now an array of length 1.
+3. One of the signers uses a smart contract account controlled by an ECDSA key held in a hardware wallet, so their wallet requests an EIP-712 signature from the hardware wallet, prepending the domain of the smart contract account to `signingDomains`, which is now an array of length 2.
+4. The signer verifies the contents of the signature in their hardware wallet. They are able to see that they are signing a message intended for a particular dapp domain, on behalf of their smart contract account (closest signing domain), as a member of the multisig (furthest signing domain).
+
+#### Encoding of data to be signed
+
+In the presence of `signingDomains` the account should encode the message to be signed according to the following recursive procedure:
+
+- `encodeForSigningDomains(signingDomainSeparators : [𝔹²⁵⁶], verifyingDomainSeparator : 𝔹²⁵⁶, message : 𝕊) =`
+  - If `signingDomainSeparators = [first, ...others]`: `"\x19\x02" ‖ first ‖ encodeForSigningDomains(others, verifyingDomainSeparator, message)`
+  - If `signingDomainSeparators = []`: `encode(domainSeparator, message)`, where `encode` is defined by EIP-712.
+
+`signingDomainSeparators` is the array of hashes of the domains included in `signingDomains`, in the same order, computed according to EIP-712's `hashStruct`.
+
+### `authMethods`
+
+This property is an array of supported signature authentication methods, listed in the order that the verifying domain tries them.
+
+Each member of the array is an object with the following keys:
+
+- `id`: An string that identifies the method. It may be one of:
+    - `ECDSA`: ECDSA signatures by Externally Owned Accounts.
+    - `ERC-{n}`: A standard type of signature specified by an ERC. `n` must not be padded with zeros.
```

Comment:
Is there an authentication method other than ERC-1271? If 1271 is generalized enough I think this can be simplified to just `ECDSA` or `ERC-1271`.

In the case we want to make this ERC more broadly compatible, then it may make sense to define the signature algorithms supported by other networks (e.g. ed25519 in cosmos). I feel this is not the case

Reply:
>Is there an authentication method other than ERC-1271? 

Yes, ERC-6492 is another standard that is mentioned earlier in the Motivation section. It covers a scenario that ERC-1271 doesn't (predeploy contracts). I'd rather keep this generic than listing the standards that exist now.